docs/book/src/getting-started/on-tree-sitter.md

```diff
@@ -0,0 +1,122 @@
+
+For example, say you wanted to find `for` loops with an `if` statement
+as the first child. The query for this might look something like:
+
+```scheme
+(for_loop
+  .
+  (if_statement)
+)
+```
+
+The syntax for Tree-sitter queries is described in the [Tree-sitter
+documentation][tree-sitter:queries],[^support] which we won't go into
+here. However, there is one very important concept in Tree-sitter
+queries, as far as Topiary is concerned: what Tree-sitter calls "capture
+names".
+
+When you write a query and you want to capture a specific node in that
+query, rather than the entire subtree, you can annotate the node with a
+"capture name". These are represented with an `@` character, followed by
+an identifier. (To stretch our regular expression analogy, these would
+be similar to named groups.)
+
+In our above example, say we actually only care about `if` statements
+that appear as the first child of `for` loops; we don't care about the
+rest of the subtree. Then we could add a capture name as follows:
+
+```scheme
+(for_loop
+  .
+  (if_statement) @my_important_node
+)
+```
+
+It's this mechanism that Topiary relies on to perform formatting. That
+is, queries are used to identify syntactic constructions of interest,
+then specific capture names -- for which Topiary [defines particular
+semantics](../reference/capture-names/index.md) -- apply respective
+formatting to those nodes within that and every matching subtree.
+
+Say, for example, we always want an `if` statement that appears as the
+first child of a `for` loop to be on a new line. For this, we could use
+the [`@prepend_hardline` capture name](../reference/capture-names/vertical-spacing.md#append_hardline--prepend_hardline):
+
+```scheme
+(for_loop
+  .
+  (if_statement) @prepend_hardline
+)
+```
+
+This is the essence of Topiary.
+
+<div class="warning">
+This example is contrived for illustrative purposes. The queries you
+would actually write would probably be much more general, so you don't
+have to enumerate every syntactic possibility in which a line break
+needs to be inserted. However, as Tree-sitter is so flexible, if
+exceptional rules like this exist, they can easily be accommodated.
+</div>
+
+### Topiary language query files
+
+A language query file, usually given then `.scm` extension, is a
+collection of such queries with appropriate formatting capture names.
+Taken in aggregate, when applied to source code files by Topiary, they
+define a formatting style for the language in question.
+
+<div class="warning">
+Topiary makes no assumption about a language's token separators. When
+it parses any input source into a collection of nodes, it will only
+apply formatting that has been explicitly defined with capture nodes, in
+a query file. This can result in any unspecified nodes losing their
+token separators (e.g., spacing) after formatting. That is, nodes can be
+"squashed" together, which can change (or even break) the code's
+semantics.
```

Comment:
Probably this deserves a link to an example.

Reply:
Added as #910